### ISSUE TITLE
[Research] Upgrade to emotion2vec+ Large and Add Whisper Encoder Dual-Use for Free Emotion Signal

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#36]
## What

Two voice model upgrades:
1. **emotion2vec+ large** (~300M params) — upgrade from base for better accuracy
2. **Whisper encoder dual-use** — if using Whisper for transcription, extract emotion signal from encoder hidden states at zero additional compute

### emotion2vec+ large
- Current: emotion2vec+ base (~90M params) — already SOTA among SSL models
- Large variant (~300M params): higher accuracy, same FunASR API
- Only worthwhile if server has sufficient RAM (Railway may be tight)

### Whisper encoder dual-use
- Feb 2026 paper: Whisper encoder contains rich paralinguistic information
- Fine-tuned Whisper-Large-V3 on RAVDESS+SAVEE+TESS: **92% accuracy** across 7 emotions
- If we already use Whisper for dream journal transcription → emotion is a **free byproduct**
- Attention-based pooling on encoder hidden states → classification head

### emotion2vec+ large
```python
# ml/models/voice_emotion_model.py
# Change one line:
self.model = AutoModel("iic/emotion2vec_plus_large")  # was: plus_base
```
Check Railway memory limits first. If >2GB available, upgrade.

### Whisper encoder emotion
```python
# ml/models/whisper_emotion.py (NEW)

import whisper
import torch

class WhisperEmotionExtractor:
    """Extract emotion from Whisper encoder hidden states.
    
    Zero additional compute if already running Whisper for ASR.
    Just tap into intermediate encoder layers.
    """
    def __init__(self):
        self.whisper_model = whisper.load_model("large-v3")
        self.emotion_head = nn.Linear(1280, 7)  # 7 emotions
        # Fine-tune emotion_head on RAVDESS/IEMOCAP
    
    def transcribe_and_classify(self, audio):
        """Single forward pass: get both transcription + emotion."""
        # Run Whisper encoder
        mel = whisper.log_mel_spectrogram(audio)
        encoder_output = self.whisper_model.encoder(mel)
        
        # ASR path (normal Whisper)
        transcription = self.whisper_model.decode(encoder_output)
        
        # Emotion path (attention pooling on encoder states)
        pooled = attention_pool(encoder_output)
        emotion_probs = softmax(self.emotion_head(pooled))
        
        return transcription, emotion_probs
```

### Files to Modify
- `ml/models/voice_emotion_model.py` — Upgrade emotion2vec+ to large
- `ml/models/whisper_emotion.py` (NEW) — Whisper encoder emotion extraction
- `ml/api/routes.py` — Wire Whisper emotion into voice analysis

### Expected Impact
- emotion2vec+ large: **+3-5%** over base on emotion accuracy
- Whisper dual-use: **free emotion signal** alongside transcription (92% on controlled data)

## Complexity
LOW (emotion2vec+ upgrade) / MEDIUM (Whisper dual-use requires fine-tuning)

## References
- emotion2vec+: https://huggingface.co/emotion2vec
- Whisper SER (Feb 2026): https://arxiv.org/abs/2602.06000
- Fine-tuned Whisper: https://huggingface.co/firdhokk/speech-emotion-recognition-with-openai-whisper-large-v3
